From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Giuseppe Maggio <jertlok@proton.me>
Date: Sat, 22 Aug 2026 23:30:00 +0200
Subject: [PATCH] arm64: dts: qcom: msm8998: let the GPU vote for DDR bandwidth

Wire the gfx-mem path and give every GPU OPP a bandwidth, so the OPP
core scales DDR alongside the core clock, as sdm630/sdm660 do for
their a5xx. The values are the downstream kgsl
qcom,msm-bus,vectors-KBps entries selected by each power level's
bus-freq index (msm8998-gpu.dtsi; msm8998-v2.dtsi power levels - the
mainline OPPs match them one to one). Bandwidth is given as
opp-avg-kBps as well as opp-peak-kBps because this RPM's DDR plan
follows the average-bandwidth arb votes. Active-only tag for the same
sleep-set reason as the BWMON.

Assisted-by: LLM
Signed-off-by: Giuseppe Maggio <jertlok@proton.me>
---
 arch/arm64/boot/dts/qcom/msm8998.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8998.dtsi b/arch/arm64/boot/dts/qcom/msm8998.dtsi
index b47f757d6b69..893cb6a4be87 100644
--- a/arch/arm64/boot/dts/qcom/msm8998.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8998.dtsi
@@ -2530,6 +2530,9 @@ adreno_gpu: gpu@5000000 {
 
 			interrupts = <GIC_SPI 300 IRQ_TYPE_LEVEL_HIGH>;
 			iommus = <&adreno_smmu 0>;
+			interconnects = <&bimc MASTER_GRAPHICS_3D RPM_ACTIVE_TAG
+					 &bimc SLAVE_EBI RPM_ACTIVE_TAG>;
+			interconnect-names = "gfx-mem";
 			operating-points-v2 = <&gpu_opp_table>;
 			power-domains = <&rpmpd MSM8998_VDDMX>;
 			status = "disabled";
@@ -2562,6 +2565,8 @@ gpu_opp_table: opp-table {
 				compatible = "operating-points-v2";
 				opp-710000097 {
 					opp-hz = /bits/ 64 <710000097>;
+					opp-avg-kBps = <14432000>;
+					opp-peak-kBps = <14432000>;
 					opp-microvolt = <1056000 1056000 1100000>;
 					opp-level = <RPM_SMD_LEVEL_TURBO>;
 					opp-supported-hw = <0xff>;
@@ -2569,6 +2574,8 @@ opp-710000097 {
 
 				opp-670000048 {
 					opp-hz = /bits/ 64 <670000048>;
+					opp-avg-kBps = <14432000>;
+					opp-peak-kBps = <14432000>;
 					opp-microvolt = <1056000 1056000 1100000>;
 					opp-level = <RPM_SMD_LEVEL_TURBO>;
 					opp-supported-hw = <0xff>;
@@ -2576,6 +2583,8 @@ opp-670000048 {
 
 				opp-596000097 {
 					opp-hz = /bits/ 64 <596000097>;
+					opp-avg-kBps = <12440000>;
+					opp-peak-kBps = <12440000>;
 					opp-microvolt = <900000 900000 1100000>;
 					opp-level = <RPM_SMD_LEVEL_NOM>;
 					opp-supported-hw = <0xff>;
@@ -2583,6 +2592,8 @@ opp-596000097 {
 
 				opp-515000097 {
 					opp-hz = /bits/ 64 <515000097>;
+					opp-avg-kBps = <12440000>;
+					opp-peak-kBps = <12440000>;
 					opp-microvolt = <900000 900000 1100000>;
 					opp-level = <RPM_SMD_LEVEL_NOM>;
 					opp-supported-hw = <0xff>;
@@ -2590,6 +2601,8 @@ opp-515000097 {
 
 				opp-414000000 {
 					opp-hz = /bits/ 64 <414000000>;
+					opp-avg-kBps = <8136000>;
+					opp-peak-kBps = <8136000>;
 					opp-microvolt = <828000 828000 1100000>;
 					opp-level = <RPM_SMD_LEVEL_SVS>;
 					opp-supported-hw = <0xff>;
@@ -2597,6 +2610,8 @@ opp-414000000 {
 
 				opp-342000000 {
 					opp-hz = /bits/ 64 <342000000>;
+					opp-avg-kBps = <6144000>;
+					opp-peak-kBps = <6144000>;
 					opp-microvolt = <828000 828000 1100000>;
 					opp-level = <RPM_SMD_LEVEL_LOW_SVS>;
 					opp-supported-hw = <0xff>;
@@ -2604,6 +2619,8 @@ opp-342000000 {
 
 				opp-257000000 {
 					opp-hz = /bits/ 64 <257000000>;
+					opp-avg-kBps = <3296000>;
+					opp-peak-kBps = <3296000>;
 					opp-microvolt = <828000 828000 1100000>;
 					opp-level = <RPM_SMD_LEVEL_MIN_SVS>;
 					opp-supported-hw = <0xff>;
